### PR TITLE
Add support for sass indented syntax

### PR DESCRIPTION
**Pull request by paradoxxxzero**
Add support for indented syntax (import 'style.sass!scss')
Internal sass imports are not yet supported as per https://github.com/sass/libsass/issues/1695

Please review.

### CODE REVIEW

**Comment by manveru** [comment]
I tried using your version with `jspm install scss=github:paradoxxxzero/plugin-sass-1` and then `import 'app.css!scss'`.
Right now I'm getting a syntax error that basically says it's trying to parse the file as `SCSS` and not `SASS`. Any idea what I might be doing wrong?

Also weird that a project called `plugin-sass` doesn't actually support `SASS` syntax, so thanks for helping to fix that.

File: src/sass-inject.js [hunk 1]
```diff
@@ -60,9 +62,13 @@ const compile = scss => {
 
 export default load => {
   urlBase = load.address;
+  let indentedSyntax = urlBase.split('.').slice(-1)[0] == 'sass'
```
Comment (screendriver) [inline src/sass-inject.js]: You can use the shiny new [String.prototype.endsWith()](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/endsWith) method here.

File: src/sass-builder.js [hunk 2]
```diff
@@ -33,6 +34,8 @@ const loadFile = path => {
 
 // intercept file loading requests (@import directive) from libsass
 sass.importer((request, done) => {
+// Currently only supporting scss imports due to
+// https://github.com/sass/libsass/issues/1695
```
Comment (screendriver) [inline src/sass-builder.js]: Please indent 2 characters here

File: src/sass-builder.js [hunk 3]
```diff
@@ -46,26 +49,31 @@ sass.importer((request, done) => {
...
+  const stubDefines = loads.map(load => {
+    return `${(compileOpts.systemGlobal || 'System')}\.register('${load.name}', [], false, function() {});`;
+  }).join('\n');
+
+  const compile_promise = load => {
+    return new Promise((resolve, reject) => {
       urlBase = load.address;
+      const options = {
+        style: sass.style.compressed,
+        indentedSyntax: urlBase.split('.').slice(-1)[0] == 'sass'
```
Comment (screendriver) [inline src/sass-builder.js]: You can use the shiny new [String.prototype.endsWith()](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/endsWith) method here.

**Comment by screendriver** [comment]
Thank you very much for the pull request! :+1: 
Could you please fix all [ESLint](http://eslint.org) errors?

**Comment by paradoxxxzero** [comment]
@manveru Try `jspm install scss=github:paradoxxxzero/plugin-sass-1@sass` it works when specifying a branch (otherwise it uses the official repo maybe due to the package.json)

@screendriver I'll fix it asap

**Comment by screendriver** [comment]
By the way: you can do a simple `$ gulp` or `$gulp lint` to see all errors :wink:

**Comment by paradoxxxzero** [comment]
Thanks

BTW i was thinking, is there anything that guarantee the style insertion order in sass-inject ?

I mean if I 

```js
System.import('style-1.scss!');
System.import('style-2.scss!');
System.import('style-3.scss!');
```

shouldn't be the transpiled style inserted in this specific order ? (This is guaranteed in sass-builder)

**Comment by screendriver** [comment]
As far as I know is the order not guaranteed. So a

```js
const one = System.import('style-1.scss!');
const two = System.import('style-2.scss!');
const three = System.import('style-3.scss!');
Promise.all([ one, two, three ]);
```

loads all three, but **maybe** not in order because the first file could be larger as the second one and therefore the underlying AJAX request for the second one returns earlier as the first one. If you want a order guarantee you have to import it sequential 

```js
System.import('style-1.scss!')
.then(() => System.import('style-2.scss!'))
.then(() => System.import('style-3.scss!'));
```

**Comment by screendriver** [comment]
I've made a [v0.10](https://github.com/mobilexag/plugin-sass/releases/tag/v0.1.0) release for that.

**Comment by paradoxxxzero** [comment]
Thanks for the release.

I'm using es6 imports (no promises) and I think import order should be guaranteed.

I will look into that.

**Comment by screendriver** [comment]
If you are using the synchronous ES2015 imports like

```js
import 'one';
import 'two';
import 'three';
```

then the import order should be guaranteed because the mechanism there is slightly different.

**Comment by paradoxxxzero** [comment]
Ok thanks didn't know that :)

**Comment by j-walker23** [comment]
Thanks for this guys, really appreciated. 
Looking at the issue in libsass with custom imports and `.sass`, they say many times to use `sass2scss`, but i have no idea how to. Is it possible on the client to do what they are saying for the work around?

**Comment by screendriver** [comment]
[sass2scss](https://github.com/mgreter/sass2scss) is written in C++ and can't be used at runtime (but at bundling time) in the browser.

**Comment by mgreter** [comment]
Opened a [feature request with node-sass] [1], but I also found https://github.com/jitcoder/node-sass2scss (and https://github.com/paazmaya/grunt-sass2scss).

[1]: https://github.com/sass/node-sass/issues/1417

**Pull request merged**